### PR TITLE
restish: init at 0.14.0

### PR DESCRIPTION
**Pull request by Fumesover**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Go tests work on x86_64-linux, and the binary have been extensively used by me and is confirmed working.

Can be backported without concerns

### CODE REVIEW

File: pkgs/applications/misc/restish/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+buildGoModule rec {
+  pname = "restish";
+  version = "0.14.0";
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "danielgtaylor";
+    repo = pname;
+    sha256 = "sha256-AzZwu9NjRRxBu4RCa8Pzi4yg6g1SIiE8V1VC2SqLiKk=";
```
Comment (sikmir) [inline pkgs/applications/misc/restish/default.nix:14]:
```suggestion
    hash = "sha256-AzZwu9NjRRxBu4RCa8Pzi4yg6g1SIiE8V1VC2SqLiKk=";
```

File: pkgs/applications/misc/restish/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  vendorSha256 = "sha256-NT928HSXWbxeyT4uunKKTjIsIAGTQtZUFh26WA/wH8Q=";
```
Comment (sikmir) [inline pkgs/applications/misc/restish/default.nix:17]:
```suggestion
  vendorHash = "sha256-NT928HSXWbxeyT4uunKKTjIsIAGTQtZUFh26WA/wH8Q=";
```

File: pkgs/applications/misc/restish/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  meta = with lib; {
+    description = "CLI for interacting with REST-ish HTTP APIs with some nice features built-in";
+    homepage = "https://rest.sh/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ fumesover ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (sikmir) [inline pkgs/applications/misc/restish/default.nix:28]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/misc/restish/default.nix (hunk 2 above)
Comment (figsoda) [reply pkgs/applications/misc/restish/default.nix:17]: this is not necessary

File: pkgs/applications/misc/restish/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/applications/misc/restish/default.nix:14]: this is not necessary

File: pkgs/applications/misc/restish/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    rev = "v${version}";
+    owner = "danielgtaylor";
+    repo = pname;
+    hash = "sha256-AzZwu9NjRRxBu4RCa8Pzi4yg6g1SIiE8V1VC2SqLiKk=";
+  };
+
+  vendorHash = "sha256-NT928HSXWbxeyT4uunKKTjIsIAGTQtZUFh26WA/wH8Q=";
+
```
Comment (figsoda) [inline pkgs/applications/misc/restish/default.nix:18]:
```suggestion

  ldflags = [ "-s" "-w" "-X=main.version=${version}" ];
  
```

Comment (sikmir) [reply pkgs/applications/misc/restish/default.nix:18]:
```suggestion

  passthru.tests.version = testers.testVersion {
    package = finalAttrs.finalPackage;
  };
```

Comment (figsoda) [reply pkgs/applications/misc/restish/default.nix:18]: Please apply my change too

File: pkgs/applications/misc/restish/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  meta = with lib; {
+    description = "CLI for interacting with REST-ish HTTP APIs with some nice features built-in";
+    homepage = "https://rest.sh/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ fumesover ];
+    platforms = platforms.unix;
```
Comment (sikmir) [inline pkgs/applications/misc/restish/default.nix:31]:
```suggestion
```

File: pkgs/applications/misc/restish/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
```
Comment (sikmir) [inline pkgs/applications/misc/restish/default.nix:4]:
```suggestion
, testers
, restish
}:
```

**Comment by waldyrious** [comment]
I'm interested in this as well. Is there anything I can do to help this PR move forward? @Fumesover if you'd like, and grant me the necessary permissions, I could apply the requested changes and rebase the branch.

**Comment by leroyshirto** [comment]
I too am interesting in using this package. Has there been any movement since?

**Pull request closed without merging**